### PR TITLE
[TESTS] Create e2e tests for `privacy links` and `network switch`

### PR DESCRIPTION
**Pull request by hetvart**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

### Summary:
PR covers e2e tests described in [the issue](https://github.com/status-im/status-react/issues/5724).
Added tests:
- privacy policy is accessible (testrail id is 4769)
- can switch network (testrail id is 690)

[comment]: # (Summarise the problem and how the pull request solves it)

### Review notes:
New view added:
_about_view.py_ describing the view Profile -> About

**NOTE:** When verifying privacy policy link we ensure that Privacy Policy summary title appears on the page opened in mobile browser (see screen)

privacy_policy_summary

#### 1. Privacy policy is accessible

|   | Step                                                                                          | Expected result                                                                                   |
|---|-----------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------|
| 1 | Open status and check Privacy policy is visible on signup                                     | The privacy policy is visible on signup                                                           |
| 2 | Click the privacy policy link                                                                 | Tapping privacy link opens in a browser (link >> https://www.iubenda.com/privacy-policy/45710059) |
| 3 | Open status - create account - log out - other accounts. Check Privacy is visible on recovery | Privacy is visible on recovery                                                                    |
| 4 | Click the privacy policy link                                                                 | Tapping privacy link opens in a browser (link >> https://www.iubenda.com/privacy-policy/45710059) |
| 5 | Open status - tap profile -> about. Check Privacy visible on the profile tab                  | Privacy visible on the profile tab                                                                |
| 6 | Tap the privacy policy item                                                                   | Tapping privacy link opens in a browser (link >> https://www.iubenda.com/privacy-policy/45710059) |

#### 2. Can switch network

|   | Step                                                                           | Expected result                                      |
|---|--------------------------------------------------------------------------------|------------------------------------------------------|
| 1 | Go to Profile, enable dev mode and switch network to Mainnet with Upstream RPC | app warns that restart is needed                     |
| 2 | agree to restart and login                                                     | go to Profile screen and verify the selected network |

status: ready 

<blockquote><div>iubenda</div><div><strong><a href="https://www.iubenda.com/privacy-policy/45710059">Privacy Policy of Status Mobile</a></strong></div><div>This Application collects some Personal Data from its Users.</div></blockquote>

### CODE REVIEW

File: test/appium/tests/base_test_case.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def capabilities_local(self):
         desired_caps['deviceName'] = 'nexus_5'
         desired_caps['platformName'] = 'Android'
         desired_caps['appiumVersion'] = '1.7.2'
-        desired_caps['platformVersion'] = '7.1'
+        desired_caps['platformVersion'] = '8.0'
```
Comment (antdanchenko) [inline test/appium/tests/base_test_case.py]: please remove the change, we do not expect to migrate on 8.0 version in scope of this PR

File: test/appium/tests/test_general_verifications.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        if signin_view.ok_button.is_element_displayed():
+            signin_view.ok_button.click()
+        signin_view.other_accounts_button.click()
+
+        if not signin_view.privacy_policy_link.is_element_displayed():
+            self.driver.fail(f'{no_link_found_error_msg} Sign in view!')
+
+        base_web_view = signin_view.privacy_policy_link.click()
+        if not base_web_view.policy_summary.is_element_displayed():
+            self.errors.append(f'{no_link_open_error_msg} Sign in view!')
```
Comment (antdanchenko) [inline test/appium/tests/test_general_verifications.py:45]: please add `self.verify_no_errors()` in the end of each test, if there is at least one `errors.append`
Without the method the test will always pass :)

**Review by antdanchenko — CHANGES_REQUESTED** [review]

File: test/appium/views/about_view.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+from views.base_element import BaseButton, BaseText
+from views.base_view import BaseView
+
+
+class PrivacyPolicyButton(BaseButton):
+    def __init__(self, driver):
+        super(PrivacyPolicyButton, self).__init__(driver)
+        self.locator = self.Locator.xpath_selector(
+            '//android.view.ViewGroup[@content-desc="privacy-policy"]/android.view.ViewGroup')
```
Comment (antdanchenko) [inline test/appium/views/about_view.py]: please, use more general xpath for all added elements
e.g. //*[@content-desc="privacy-policy"]

File: test/appium/tests/atomic/account_management/test_profile.py [hunk 4]
```diff
@@ -223,6 +223,19 @@ def test_deny_device_storage_access_changing_profile_photo(self):
         profile.select_from_gallery_button.click()
         profile.deny_button.wait_for_visibility_of_element(2)
 
+    @marks.testrail_id(690)
+    @marks.smoke_1
+    def test_user_can_switch_network(self):
+        signin_view = SignInView(self.driver)
+        self.home_view = signin_view.create_user()
```
Comment (yevh-berdnyk) [inline test/appium/tests/atomic/account_management/test_profile.py]: We don't need to create a class instance attribute here

File: test/appium/views/profile_view.py [hunk 5]
```diff
@@ -374,6 +374,24 @@ def __init__(self, driver):
...
+    def __init__(self, driver):
+        super(ActiveNetworkName, self).__init__(driver)
+        self.locator = self.Locator.text_part_selector('WITH UPSTREAM RPC')
+
+
+class AboutButton(BaseButton):
+    def __init__(self, driver):
+        super(AboutButton, self).__init__(driver)
+        self.locator = self.Locator.xpath_selector(
+            '//android.view.ViewGroup[@content-desc="about-button"]/android.view.ViewGroup')
```
Comment (yevh-berdnyk) [inline test/appium/views/profile_view.py]: It's better to use accessibility ids where it's possible, here it's 'about-button'

**Comment by hetvart** [comment]
@antdanchenko @yevh-berdnyk 
went through PR and made changes according to your comments.
Please review.

**Review by yevh-berdnyk — APPROVED** [review]

**Review by antdanchenko — APPROVED** [review]

**Comment by antdanchenko** [comment]
@hetvart please rebase the PR on `status-im/develop` in order to fix the issue with `This commit cannot be built`

**Comment by vitvly** [comment]
@hetvart Can you please GPG-sign your commits as well?

**Pull request merged**